### PR TITLE
Users/dstull/sock opt string length

### PR DESCRIPTION
**Pull request by danestull**
### What does this Pull Request accomplish?

Fetches Size of string socket options.

### Why should this Pull Request be merged?

This allows us to know how large the string value for string sock opts are before we allocate them, therefore allowing us to be more efficient. This is done by calling the GetSockOpt with a nullptr for the value reference first and this is done in the new size() method of the converter. 

### What testing has been done?

Created new test `StringSockOptData_GetSize_FetchesDataFromLibrary'
All builds and tests pass
Screen Shot 2022-03-29 at 9 45 28 AM

### CODE REVIEW

File: source/tests/unit/xnet_converters_tests.cpp [hunk 1]
```diff
@@ -601,6 +608,25 @@ TEST(XnetConvertersTests, IPv4AddrOutputConverter_ConvertToGrpc_ConvertsToAddres
...
 }
 
+TEST(XnetConvertersTests, StringSockOptData_GetSize_FetchesSizeFromLibrary)
+{
+  constexpr auto DEFAULT_SOCK_OPT_STRING_SIZE = 255;
+  auto ACTUAL_SOCK_OPT_STRING_SIZE = 12;
+  NiXnetSocketMockLibrary library;
+  auto storage = allocate_output_storage<void*, SockOptData>(&library, OptName::OPT_NAME_SO_BIND_TO_DEVICE);
+  EXPECT_EQ(DEFAULT_SOCK_OPT_STRING_SIZE, storage.string_length);
+  // Expect call GetSockOpt(socket, level, opt_name, nullptr, &string_length)
```
Comment (mgumble) [inline source/tests/unit/xnet_converters_tests.cpp:618]: not sure if this comment provides a lot, it is obvious we expect the call from line 623

**Review by mgumble — APPROVED** [review]

**Review by reckenro — APPROVED** [review]

File: generated/nixnetsocket/nixnetsocket.proto [hunk 2]
```diff
@@ -148,7 +148,7 @@ message GatewayAddress {
 
 message VirtualInterface {
   string xnet_interface_name = 1;
-  string vlane_name = 2;
+  string vlan_name = 2;
```
Comment (astarche) [inline generated/nixnetsocket/nixnetsocket.proto:151]: fyi: I already made this change.

File: source/tests/unit/xnet_converters_tests.cpp [hunk 3]
```diff
@@ -601,6 +608,25 @@ TEST(XnetConvertersTests, IPv4AddrOutputConverter_ConvertToGrpc_ConvertsToAddres
   EXPECT_EQ(ADDRESS, grpc_addr.addr());
 }
 
+TEST(XnetConvertersTests, StringSockOptData_GetSize_FetchesSizeFromLibrary)
+{
+  constexpr auto DEFAULT_SOCK_OPT_STRING_SIZE = 255;
+  auto ACTUAL_SOCK_OPT_STRING_SIZE = 12;
```
[the remark this answers is not in the document]
Comment (danestull) [reply source/tests/unit/xnet_converters_tests.cpp]: I've updated these both to just `const` not sure the exact reason constexpr was chose for several other instances, perhaps just so its set at compile time? But I see no problem with just having these both be consts

File: source/tests/unit/xnet_converters_tests.cpp (hunk 1 above)
Comment (danestull) [reply source/tests/unit/xnet_converters_tests.cpp]: removed

File: generated/nixnetsocket/nixnetsocket.proto (hunk 2 above)
Comment (danestull) [reply generated/nixnetsocket/nixnetsocket.proto:151]: Mose had mentioned you've made some updates so I'll be sure to handle any merge conflicts / remove this from mine

**Review by astarche — APPROVED** [review]

File: source/custom/xnetsocket_converters.h [hunk 4]
```diff
@@ -548,8 +549,7 @@ struct SockOptDataOutputConverter {
         output.set_data_bool(data_int == 0 ? false : true);
         break;
       }
-      case OptName::OPT_NAME_SO_BIND_TO_DEVICE:
-      case OptName::OPT_NAME_SO_ERROR: {
+      case OptName::OPT_NAME_SO_BIND_TO_DEVICE: {
```
[the remark this answers is not in the document]
Comment (danestull) [reply source/custom/xnetsocket_converters.h]: Updated these after merging should be all consistent!

**Pull request merged**